### ISSUE TITLE
Any thoughts about supporting B9PartSwitch?

### ISSUE THREAD

**Issue by SteveBenz** [Nils277/KerbalPlanetaryBaseSystems#99]
I've been developing a mod, [Progressive Colonization System](https://forum.kerbalspaceprogram.com/index.php?/topic/181852-15x~17x-progressivecolonizationsystem-life-support-colonization-for-more-fun-in-late-game/), which uses PBS parts.  It also introduces a variety of resources, and it'd certainly be nice if there were some KPBS parts that could store them.  I've also been using Station Parts Expansion Redux, which has storage based on B9Partswitch.  I'd love to see that in KPBS too.

But it's not as easy as just a ModuleManager fix to do it (well, at least if you want it to look spiffy).  They can implement a decal on the side that shows what the payload is.  I'm well-able to contribute the ModuleManager mojo and testing for the part, though I'll struggle a bit with creating the texture.

Is this a thing you'd be interested in adding?  (In my mind, the resources that KPBS would support would include Ore, EPL stuff and Community Resource Kit -- my mod would have to configure on my resources until it gains more community acceptance.)

For bonus points, I'd love it if there was a deployable storage system to allow storing larger amounts of stuff once-landed.

**Comment by Nils277** [comment]
I already have planned to add support for switchable resources to KPBS.
In fact my other mod FUR already does have that funtionality. However it uses a custom resource switching that i have developed for it.
KPBS will use the same resource switching as FUR, but i have not time at the moment to do that. Maybe in around half a year or so.

**Comment by Grimmas** [comment]
@Nils277 Do you still plan on adding the FUR part-switch code to KPBS? If you do plan to go ahead then you may wish to consider just splitting that code off into a third project that both FUR and KPBS can depend on. 

I was considering adding a few KPBS parts for switchable resources for CRP/MKS (no new models, just using the existing assets), but this gave me pause. B9 is great, and it has the ability to select any choice from a list without going through the whole collection one by one, but OTOH I also do not want to introduce a second type of switch to your repository, it would only confuse potential users (though anyone using this with MKS is already exposed to the horrors of Firespitter switches so maybe that's a moot point).

**Comment by Nils277** [comment]
@Grimmas Yes this is still planned. The folder structure of FUR was actually already designed to work with this in mind. With _000_KerbetrotterTools_ containing the common tools and modules.
All KPSB parts will eventually move into KerbetrotterLtd/PlanetaryBaseSystem so it will look like this in the end:

GameData
├━KerbetrotterLtd
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;├━000_KerbetrotterTools --->Common
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;├━FelineUtilityRover ---> FUR
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;┗━PlanetaryBaseSystem ---> KPBS
├━~~PlanetaryBaseInc~~
├━ Squad
┆
It is also planned to make the switching from KerbetrotterTools to use a list instead of iterating through all resources.
It would be good if you use the Switching Modules from FUR for the CRP/MKS parts. :-)

**Comment by Grimmas** [comment]
Submitted a patch to KPBS and to FUR just now to enable this. Nothing fancy, no model switching or anything.
